### PR TITLE
[Core] Deprecate get_shape method

### PR DESCRIPTION
**Pull request by andrei-cv**
### Details:
 - *Deprecate get_shape method*

### Tickets:
 - *CVS-148104*

### CODE REVIEW

**Review by dmatveev — APPROVED** [review]
Approved the NPUW change

File: src/core/include/openvino/core/model.hpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
     const ov::element::Type& get_output_element_type(size_t i) const;
 
     /// Return the shape of element i
-    const Shape& get_output_shape(size_t i) const;
+    Shape get_output_shape(size_t i) const;
```
Comment (olpipi) [inline src/core/include/openvino/core/model.hpp:156]: We should not change return value of these methods. Just deprecate them.
Please add deprecated annotation to all these methods like here:
https://github.com/openvinotoolkit/openvino/blob/2c9953320a575f24856cb5273d6d23c7600c880b/src/core/include/openvino/op/concat.hpp#L40
Release to remove, I suppose, is 2026.0
+
Add OPENVINO_SUPPRESS_DEPRECATED_START/OPENVINO_SUPPRESS_DEPRECATED_END in every place, where these methods are used.

**Review by olpipi — CHANGES_REQUESTED** [review]

File: src/core/include/openvino/core/node_output.hpp [hunk 2]
```diff
@@ -155,7 +155,7 @@ class OPENVINO_API Output<const Node> {
     /// \return The element type of the output referred to by this output handle.
     OV_NO_DANGLING const element::Type& get_element_type() const;
     /// \return The shape of the output referred to by this output handle.
-    OV_NO_DANGLING const Shape& get_shape() const;
+    OV_NO_DANGLING Shape get_shape() const;
```
Comment (praasz) [inline src/core/include/openvino/core/node_output.hpp:158]:
```suggestion
    Shape get_shape() const;
```
Is `OV_NO_DANGLING ` required when return shape by value (copy)?

File: src/core/include/openvino/core/node_input.hpp [hunk 3]
```diff
@@ -92,7 +92,7 @@ class OPENVINO_API Input<const Node> {
     /// \return The element type of the input referred to by this input handle.
     OV_NO_DANGLING const element::Type& get_element_type() const;
     /// \return The shape of the input referred to by this input handle.
-    OV_NO_DANGLING const Shape& get_shape() const;
+    OV_NO_DANGLING Shape get_shape() const;
```
Comment (praasz) [inline src/core/include/openvino/core/node_input.hpp:95]:
```suggestion
    Shape get_shape() const;
```
Is `OV_NO_DANGLING ` required when return shape by value (copy)?

**Pull request closed without merging**